server/src/immich/api-v1/asset/asset.service.ts

```diff
@@ -142,6 +145,73 @@ export class AssetService {
     }
   }
 
+  public async importFile(authUser: AuthUserDto, dto: ImportAssetDto): Promise<AssetFileUploadResponseDto> {
+    dto = {
+      ...dto,
+      assetPath: path.resolve(dto.assetPath),
+      sidecarPath: dto.sidecarPath ? path.resolve(dto.sidecarPath) : undefined,
+    };
+
+    const assetPathType = mime.lookup(dto.assetPath) as string
```

Comment:
We may want/need to test this lookup library for the mime types we support to make sure it correctly sends us the right one.